src/constraints/util/typedArray.ts

```diff
@@ -0,0 +1,16 @@
+export const TypedArrays = {
+	i8: (x: unknown): x is Int8Array => x instanceof Int8Array,
+	u8: (x: unknown): x is Uint8Array => x instanceof Uint8Array,
+	u8clamped: (x: unknown): x is Uint8ClampedArray => x instanceof Uint8ClampedArray,
+	i16: (x: unknown): x is Int16Array => x instanceof Int16Array,
+	u16: (x: unknown): x is Uint16Array => x instanceof Uint16Array,
+	i32: (x: unknown): x is Int32Array => x instanceof Int32Array,
+	u32: (x: unknown): x is Uint32Array => x instanceof Uint32Array,
+	f32: (x: unknown): x is Float32Array => x instanceof Float32Array,
+	f64: (x: unknown): x is Float64Array => x instanceof Float64Array,
+	bi64: (x: unknown): x is BigInt64Array => x instanceof BigInt64Array,
+	bu64: (x: unknown): x is BigUint64Array => x instanceof BigUint64Array,
+	TypedArray: (x: unknown): x is NodeJS.TypedArray => ArrayBuffer.isView(x) && !(x instanceof DataView)
```

Comment:
This should not use a NodeJS type but rather a global one (be it defined or otherwise)

Reply:
I just changed it to node bcz I tought `@types/node` will be present in browser also. Should I revert it?

Reply:
@types/node might not be present in browser envs at all, so.. ideally we'd either reuse defined types fron TS or make our own